play.pokemonshowdown.com/js/client-teambuilder.js

```diff
@@ -1792,31 +1797,76 @@
 				self.importSetButtons();
 			}, 'text');
 		},
+		updateCachedUserSets: function (format) {
+			if(this.userSets?.[format]) return;
+
+			this.userSets = this.userSets || {};
+			this.userSets[format] = this.userSets[format] || {};
+
+			var duplicateNameIndices = {};
+			for(const team of teams) {
+				if(team.format === format && team.capacity === 24) {
+					const setList = Storage.unpackTeam(team.team);
+					for(const pokemon of setList) {
+						var name = pokemon.name + " " + (duplicateNameIndices[pokemon.name] || "");
+						var sets = this.userSets[format][pokemon.species];
+						this.userSets[format][pokemon.species] = $.extend(sets || {}, {[name] : pokemon});
+						duplicateNameIndices[pokemon.name] = 1 + (duplicateNameIndices[pokemon.name] || 0);
+					}
+				}
+			}
+		},
+		clearCachedUserSetsIfNecessary: function (format) {
+			// clear cached user sets if we have just been in a box for given format
+			if(this.curTeam?.capacity === 24 && this.userSets?.[format]) {
```

Comment:
`?.` is also not ES3.

Is something wrong with our CI? It should be catching this...

Reply:
Oh, CI needs to be manually approved.